src/pot.erl

```diff
@@ -90,8 +90,8 @@ hotp(Secret, IntervalsNo, Opts) ->
     TokenLength = proplists:get_value(token_length, Opts, 6),
     Key = pot_base32:decode(Secret),
     Msg = <<IntervalsNo:8/big-unsigned-integer-unit:8>>,
-    Digest = crypto:hmac(DigestMethod, Key, Msg),
-    <<_:19/binary, Ob:8>> = Digest,
+    Digest = crypto:mac(hmac, DigestMethod, Key, Msg),
```

Comment:
OK, looks like this doesn't work with OTP 20/21, and I would like to keep that compatibility. I would suggest abstracting this to a local function and using the new `OTP_RELEASE` flag like [this](https://github.com/erlcloud/erlcloud/blob/c02b1fe5dcdd5ec13702d958003881cce745edb5/src/erlcloud_util.erl#L28-L39). Note that you have to check whether it's defined AND whether it is >= 23 as in the example above.

Reply:
ok, i have added your suggestion.
my Erlang-fu is weak, but hopefully this works